File: DaggerProject3/app/src/main/java/com/example/daggerproject3/model/MediaTek.java
package com.example.daggerproject3.model;

import android.util.Log;

import javax.inject.Inject;
import javax.inject.Named;

public class MediaTek implements Processor {

    int clockSpeed;
    int core;

    @Inject
    public MediaTek(@Named("clockSpeed")int clockSpeed, @Named("core")int core) {
        this.clockSpeed=clockSpeed;
        this.core=core;
        Log.i("mobile", "MediaTek: start with clockSpeed "+clockSpeed+" core: "+core);
    }

    @Override
    public void start() {
        Log.i("mobile", "MediaTek start: ");
    }
}

/**
 *The parameter in @Named must be same as provided in the Component interface
 **/

File: DaggerProject3/app/src/main/java/com/example/daggerproject3kotlin/model/MediaTek.kt
package com.example.daggerproject3kotlin.model

import android.util.Log
import javax.inject.Inject
import javax.inject.Named
import kotlin.properties.Delegates

class MediaTek @Inject constructor(@Named("clockSpeed") val clockSpeed:Int, @Named("core") val core:Int) : Processor {

    init {
        Log.i("mobile", "mediaTek created: with clockSpeed: $clockSpeed core: $core")
    }
}